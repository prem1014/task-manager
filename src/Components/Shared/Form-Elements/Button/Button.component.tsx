import React from 'react';

interface IButtonProps {
    children: any;
    clicked(event: any): any
    type: any
    cssClassName?: string;
    disabled?: boolean;
}

const Button: React.FC<IButtonProps> = (props: IButtonProps) => {
    return (
        <button
            className={props.cssClassName ? props.cssClassName : 'btn'}
            type={props.type}
            onClick={props.clicked}
            disabled={props.disabled}
        >{props.children}</button>
    )
}

export default Button;