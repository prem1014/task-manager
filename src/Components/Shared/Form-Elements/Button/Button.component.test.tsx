import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Button from './Button.component';

describe('Button component test suite', () => {
    Enzyme.configure({ adapter: new Adapter() });
    let componentWrapper: any;
    const inputProps = {
        children: 'Save',
        clicked: jest.fn(),
        type: 'button'
    }

    beforeEach( () => {
        componentWrapper = shallow(<Button {...inputProps}/>);
    });

    it('Button component snapshot test', () => {
        expect(componentWrapper).toMatchSnapshot();
    });
});