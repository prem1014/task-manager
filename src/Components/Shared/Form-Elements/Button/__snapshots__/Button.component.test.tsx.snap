// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test suite Button component snapshot test 1`] = `ShallowWrapper {}`;
